src/bin/spammer.rs

```diff
@@ -46,9 +44,29 @@ pub async fn main() -> saito_rust::Result<()> {
         let client = reqwest::Client::new();
         // sleep(Duration::from_millis(5000));
         loop {
-            let transactions =
-                generate_transactions(publickey, privatekey, txs_to_generate, bytes_per_tx);
-            for tx in transactions.iter() {
+
+            let mut transactions: Vec<Transaction> = vec![];
+
+	    for _i in 0..txs_to_generate {
```

Comment:
Overall, PR is looking good. One thing I'll request before merging this is to fix the formatting. You can see in files changed how poor the formatting can get in a bunch of places. I'm sure it looks fine in your dev environment, but for everyone else, the code becomes messy and strange. I'd like to request that you fix the spacing in your environment so that we can have proper spacing in the repo.

Reply:
I've noticed these issues. The problems were especially severe in the mempool.run() function and the GenerateTransaction logic that has now been moved into that spammer class. I suspect the "cargo fmt" formatter was struggling with the tokio/async loop internals, as well as cases within a function where we have code that looks like this -- with read / write access compartmentalized so that the lock releases quickly:

   let publickey;

  {
      let wallet = wallet_lock.write().await;
      publickey = wallet.get_publickey();
 }

The problems may also be getting caused by the Github web-interface for sorting out merges, since when the files load in that web interface they are often mangled in terms of how they display and how the different sections align. I've been using that to handle those fixes and can switch to another approach if there is something better.

At this point I've cleaned up all of those files several times. It's a bit of a disinhibitor to run "cargo fmt" when you know you might have to re-indent a lengthy function. I'm happy for us to run cargo fmt by default on code. What I try to do is clean up code when I'm looking at it in VIM and it is badly aligned. If the issue is others having a different standard for how many spaces are in a tab (I believe VIM is 8) I would be happy to switch over.

Otherwise, the best way to avoid the problem would seem to be not having large changes that sweep over multiple classes and require manual merge.